### PR TITLE
SPARKC-71: Adds suggestions for other keyspaces and tables

### PR DESCRIPTION
**Pull request by RussellSpitzer**

### CODE REVIEW

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala [hunk 1]
```diff
@@ -276,7 +276,37 @@ class CassandraRDD[R] private[connector] (
...
+              val errorString = possibleTables.mkString("\n")
+              throw new IOException(
+                s"""Couldn't find: $keyspaceName.$tableName
+                    |Other tables exist with different cases
+                    |Did you mean :
+                    |$errorString""".stripMargin)
+            } else {
+              val errorString = keyspaceList.map(_.getName).mkString("\n")
+              throw new IOException(
+                s"""Couldtn't find: $keyspaceName.$tableName
```
Comment (bcantoni) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala]: Spelling: couldn't

Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala]: Thanks! This is why I should go to sleep when it gets dark.

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala [hunk 2]
```diff
@@ -276,7 +276,37 @@ class CassandraRDD[R] private[connector] (
...
     Schema.fromCassandra(connector, Some(keyspaceName), Some(tableName)).tables.headOption match {
       case Some(t) => t
-      case None => throw new IOException(s"Table not found: $keyspaceName.$tableName")
+      case None => {
+        val clusterMetadata = connector.withClusterDo(_.getMetadata)
+        val keyspaces = clusterMetadata.getKeyspaces
+        val ksMap = keyspaces.groupBy(_.getName.toLowerCase).toMap
+        ksMap.get(keyspaceName.toLowerCase) match {
+          case Some(keyspaceList) => {
+            val possibleTables =
```
Comment (jacek-lewandowski) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala]: Maybe we can include comparison with `StringUtils.getLevensteinDistance` or `StringUtils.getJaroWinklerDistance`? We would get the same nice behaviour as git has when you misspell some command.

File: spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/CassandraRDDSpec.scala [hunk 3]
```diff
@@ -627,4 +632,28 @@ class CassandraRDDSpec extends FlatSpec with Matchers with SharedEmbeddedCassand
     result should have length 0
   }
 
+  it should "suggest mixed case keyspaces if they exist" in {
+    try {
```
Comment (pkolaczk) [inline spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/CassandraRDDSpec.scala]: Matchers contain a nice construct for this:
val exception = the [Exception] thrown by { some code } 
exception.message should ...

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala [hunk 4]
```diff
@@ -276,7 +276,37 @@ class CassandraRDD[R] private[connector] (
...
+                    |Did you mean:
+                    |$errorString""".stripMargin)
+            }
+          }
+          case None => {
+            throw new IOException(s"Table not found: $keyspaceName.$tableName")
+          }
+        }
+      }
     }
```
Comment (pkolaczk) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala]: Abstract this logic out to the Schema object or add methods for obtaining a list of similar keyspaces / tables, etc. and just call it from here. This is because this code is not really related to RDD.

**Comment by jacek-lewandowski** [comment]
+1

**Comment by jacek-lewandowski** [comment]
Please rebase it and test against Scala 2.11

**Comment by RussellSpitzer** [comment]
Rebased all tests passing

**Comment by jacek-lewandowski** [comment]
I also run all tests and it looks good. Merge it if you are not afraid of SPRAKC-25 :smiley:

**Comment by pkolaczk** [comment]
+1; just minor stylistic things
